x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/Pivot.java

```diff
@@ -41,14 +41,6 @@ public Pivot(Source source, LogicalPlan child, Expression column, List<NamedExpr
         this.aggregates = aggregates;
     }
 
-    private static Expression withQualifierNull(Expression e) {
-        if (e instanceof Attribute) {
-            Attribute fa = (Attribute) e;
-            return fa.withQualifier(null);
-        }
-        return e;
-    }
-
     @Override
     protected NodeInfo<Pivot> info() {
```

Comment:
I haven't reviewed the previous PR so there's not much to do at this point.
I find the transformation inside `info` an ugly hack - from a simple copy it became transformer that overrides qualifiers.
What if the relationship has qualifiers in it?